specs/solid-indexing/index.html

```diff
@@ -102,13 +356,16 @@ <h2>Indexes</h2>
     <section>
       <h3>General indexes</h3>
 
-      <p>An index is a RDF document [[RDF11-CONCEPTS]] of type <code>idx:Index</code> containing entries which point to
-        instances conforming to a particular shape. The shape that the index or an entry is targeting is expressed with the
-        <code>idx:hasShape</code> predicate.
-      </p>
+      <p>An index is a RDF document [[RDF11-CONCEPTS]] of type <code>idx:Index</code>. It contains entries of type <code>idf:IndexEntry</code>, each one 
+        being linked to a shape with the <code>idx:hasShape</code> predicate.</p>
+        
+      <p>An entry MUST refer to either an instance or to a sub index with the respective predicates <code>idx:hasTarget</code> and <code>idx:hasSubIndex</code>.</p>
+
+      <p>An instance is...</p>
+
+      <p>A sub index is...</p>
 
-      <p>An entry is of type <code>idx:IndexEntry</code> and targets a resource using the <code>idx:hasTarget</code>
-        predicate. The resource the entry is targeting MUST conform to the shape of the index.</p>
+      <p>A shape is...</p>
```

Comment:
suggest to copy the definition from SHACL (with a reference).

Maybe this is the place to explain that this spec recommends a very small subset of shapes.

Interestingly, we could write a SHACL shape of the SHALC shapes we allow :-)